### DISCUSSION TITLE
Set error message when using a function to validate input

### DISCUSSION THREAD

**Discussion by rjrodriguezalvarez97** [react-hook-form/react-hook-form#5126] [Q&A] [answered]
Hey everybody,
First of all, thanks for this amazing lib.
I'm trying to validate an input using a function as we can see in the [docs](https://react-hook-form.com/api/useform/register).
If I just give the function to `validate` property it works but if it fails doesn't show any message but gives focus to the input.
```js
registerOptions={
          validate: (value) => value === "1",
          message: "custom message if validation fails",
        }
```
I tried with something similar getting inspired by the `required` validator but this breaks the app.
```js
registerOptions = {
          validate: {
            value: (inputValue) =>  inputValue === "1";
            message: "custom message if validation fails",
          },
        }
```

So my question is: is it possible to give a custom message when validation fails if I'm using a function to validate the input? If so, how can I do it?

Thanks in advance :)

**Comment by bluebill1049** [comment, marked as answer]
Yes, please refer to the doc for validate function. You can return string
